### PR TITLE
Run validation on pack ignore modifications

### PR DESCRIPTION
**Pull request by YuvHayun**
## Related Issues
fixes: [CIAC-7287](https://jira-hq.paloaltonetworks.local/browse/CIAC-7287)

## Description
**Validate** will now validate items which were edited in .pack-ignore.

### CODE REVIEW

File: demisto_sdk/commands/validate/validate_manager.py [hunk 1]
```diff
@@ -1968,12 +1969,60 @@ def validate_job(self, structure_validator, pack_error_ignore_list):
 
         return is_valid
 
+    def get_all_files_edited_in_pack_ignore(self, modified_files):
```
Comment (JudahSchwartz) [inline demisto_sdk/commands/validate/validate_manager.py:1972]: add a docstring for what the function does

File: demisto_sdk/commands/validate/validate_manager.py [hunk 2]
```diff
@@ -1968,12 +1969,60 @@ def validate_job(self, structure_validator, pack_error_ignore_list):
...
+        all_files = self.git_util.get_all_files()
+        all_files_edited_in_pack_ignore = set()
+        for file_path in modified_files:
+            # handle renamed files
+            if isinstance(file_path, tuple):
+                old_file_path = file_path[0]
+                file_path = file_path[1]
+
+            else:
+                old_file_path = file_path
```
Comment (JudahSchwartz) [inline demisto_sdk/commands/validate/validate_manager.py:1982]: is this idiom used elsewhere? I dont think its very intuitive. Can we make this a function, just for documentations sake?

File: demisto_sdk/commands/validate/validate_manager.py [hunk 3]
```diff
@@ -1968,12 +1969,60 @@ def validate_job(self, structure_validator, pack_error_ignore_list):
...
+                ) == set(value):
+                    files_to_test.add(key)
+            for key, value in file_content.items():
+                if not (section_values := old_file_content.get(key, [])) or not set(
+                    section_values
+                ) == set(value):
+                    files_to_test.add(key)
+
+            def is_relevant_file(file, pack_name):
+                return len(file.parts) > 2 and file.parts[1] == pack_name
```
Comment (JudahSchwartz) [inline demisto_sdk/commands/validate/validate_manager.py:2004]: what are 2 and 1? Best not to use magic numbers

File: demisto_sdk/commands/validate/validate_manager.py [hunk 4]
```diff
@@ -1968,12 +1969,60 @@ def validate_job(self, structure_validator, pack_error_ignore_list):
...
+            if not file_path.endswith(".pack-ignore"):
+                continue
+            old_file_content = get_remote_file(old_file_path, tag="master")
+            config = ConfigParser(allow_no_value=True)
+            config.read_string(old_file_content)
+            old_file_content = self.get_error_ignore_list(config=config)
+            pack_name = get_pack_name(str(file_path))
+            file_content = self.get_error_ignore_list(pack_name)
+            files_to_test = set()
+            for key, value in old_file_content.items():
```
Comment (JudahSchwartz) [inline demisto_sdk/commands/validate/validate_manager.py:1992]: we're adding these to the same list and the loops are the same, we can do 
for key, value in old_file_content.items() + file_content.items()

File: demisto_sdk/commands/validate/validate_manager.py [hunk 5]
```diff
@@ -1968,12 +1969,60 @@ def validate_job(self, structure_validator, pack_error_ignore_list):
...
+            for key, value in file_content.items():
+                if not (section_values := old_file_content.get(key, [])) or not set(
+                    section_values
+                ) == set(value):
+                    files_to_test.add(key)
+
+            def is_relevant_file(file, pack_name):
+                return len(file.parts) > 2 and file.parts[1] == pack_name
+
+            all_files_mapper = {
```
Comment (JudahSchwartz) [inline demisto_sdk/commands/validate/validate_manager.py:2006]: how does this tell us that the file was changed in the ignore?

File: demisto_sdk/commands/validate/validate_manager.py (hunk 3 above)
Comment (YuvHayun) [reply demisto_sdk/commands/validate/validate_manager.py:2004]: Would this do?
```suggestion
                return len(file.parts) > 2 and file.parts[0] == "Packs" and file.parts[1] == pack_name
```

File: demisto_sdk/commands/validate/validate_manager.py (hunk 4 above)
Comment (YuvHayun) [reply demisto_sdk/commands/validate/validate_manager.py:1992]: But we want to ensure that when we run on old we compare against new and the other way around.

File: demisto_sdk/commands/validate/validate_manager.py (hunk 5 above)
Comment (YuvHayun) [reply demisto_sdk/commands/validate/validate_manager.py:2006]: If in the for loops above we test the content of the old pack-ignore against the new one, this is where we list all the items that there section in the pack-ignore was modified.
The idea of 'is_relevant_file' is to filter all the files that exist in the Pack so we'll avoid cases we attempt to test several files with similar name unrelated to the modified pack.

File: demisto_sdk/commands/validate/validate_manager.py (hunk 2 above)
Comment (YuvHayun) [reply demisto_sdk/commands/validate/validate_manager.py:1982]: It's used in several places around the **validate_manager**.
I'll export it to an external function.

File: demisto_sdk/commands/validate/validate_manager.py (hunk 4 above)
Comment (JudahSchwartz) [reply demisto_sdk/commands/validate/validate_manager.py:2007]: ah ok missed that

File: demisto_sdk/commands/validate/validate_manager.py (hunk 3 above)
Comment (JudahSchwartz) [reply demisto_sdk/commands/validate/validate_manager.py:2019]: im saying that what does it mean the file.parts > 2 ? and what is the first/second part?

Comment (YuvHayun) [reply demisto_sdk/commands/validate/validate_manager.py:2019]: The given file argument is some file_path from content repo.
We want to ensure that this file is related to the relevant pack so we're checking that it is under Packs/pack_name

Comment (YuvHayun) [reply demisto_sdk/commands/validate/validate_manager.py:2019]: I'll add the other connection and some doc strings.

File: demisto_sdk/commands/validate/validate_manager.py [hunk 6]
```diff
@@ -1968,29 +1969,98 @@ def validate_job(self, structure_validator, pack_error_ignore_list):
...
+                    section_values
+                ) == set(value):
+                    files_to_test.add(key)
+            for key, value in file_content.items():
+                if not (section_values := old_file_content.get(key, [])) or not set(
+                    section_values
+                ) == set(value):
+                    files_to_test.add(key)
+
+            def is_relevant_file(file, pack_name):
```
Comment (dorschw) [inline demisto_sdk/commands/validate/validate_manager.py:2018]: add types

File: demisto_sdk/commands/validate/validate_manager.py [hunk 7]
```diff
@@ -1968,29 +1969,98 @@ def validate_job(self, structure_validator, pack_error_ignore_list):
...
+            file_path: the path to extract the old file path from.
+        """
+        if isinstance(file_path, tuple):
+            old_file_path = file_path[0]
+            file_path = file_path[1]
+
+        else:
+            old_file_path = file_path
+
+        return old_file_path, file_path
```
Comment (dorschw) [inline demisto_sdk/commands/validate/validate_manager.py:1985]:
```suggestion
        return file_path[:2] if isinstance(file_path, tuple) else (file_path, file_path)
```

File: demisto_sdk/commands/validate/validate_manager.py [hunk 8]
```diff
@@ -1968,29 +1969,98 @@ def validate_job(self, structure_validator, pack_error_ignore_list):
...
+
+    def get_all_files_edited_in_pack_ignore(self, modified_files):
```
Comment (dorschw) [inline demisto_sdk/commands/validate/validate_manager.py:1987]: missing types

File: CHANGELOG.md [hunk 9]
```diff
@@ -6,6 +6,7 @@
 * Improved **update-release-notes** logs when changes in dependent API modules are detected.
 * Reverted changes released in version 1.19.0 in lint, lint will not fail on `demisto.results`, `return_outputs` and `LOG`.
 * Updated the **generate-docs** command to use the content graph instead of the id_set file.
+* **Validate** will now validate items which were edited in .pack-ignore.
```
Comment (dorschw) [inline CHANGELOG.md:9]: does it validate the files edited, or the whole pack?

Comment (YuvHayun) [reply CHANGELOG.md:9]: Only the specific files that their .pack-ignore section was edited.

File: demisto_sdk/commands/validate/validate_manager.py [hunk 10]
```diff
@@ -2695,20 +2758,21 @@ def add_ignored_errors_to_list(self, config, section, key, ignored_errors_list):
...
-                        self.add_ignored_errors_to_list(
-                            config, section, key, ignored_errors_list[file_name]
-                        )
+            config = get_pack_ignore_content(pack_name)
+        if config:
+            # create file specific ignored errors list
+            for section in filter(
+                lambda section: section.startswith("file:"), config.sections()
+            ):
+                file_name = section[5:]
```
Comment (JudahSchwartz) [inline demisto_sdk/commands/validate/validate_manager.py:2770]: whats 5?

File: demisto_sdk/commands/validate/validate_manager.py [hunk 11]
```diff
@@ -1968,29 +1969,91 @@ def validate_job(self, structure_validator, pack_error_ignore_list):
...
+                    files_to_test.add(key)
+
+            def is_relevant_file(file: Path, pack_name: str) -> bool:
+                """
+                Return wether the given file is under the given pack
+                Args:
+                    file: The file to check.
+                    pack_name: The name of the pack we want to ensure the given file is under.
+                """
+                return (
```
Comment (JudahSchwartz) [inline demisto_sdk/commands/validate/validate_manager.py:2018]: do we not have another function that checks if a file is in a pack/ it doesnt come into this function only under a certain pack?

File: demisto_sdk/commands/validate/validate_manager.py [hunk 12]
```diff
@@ -1968,29 +1969,91 @@ def validate_job(self, structure_validator, pack_error_ignore_list):
...
     def validate_modified_files(self, modified_files):
         logger.info(
             "\n[cyan]================= Running validation on modified files =================[/cyan]"
         )
         valid_files = set()
-        for file_path in modified_files:
+        all_files_edited_in_pack_ignore = self.get_all_files_edited_in_pack_ignore(
+            modified_files
+        )
+        # modified_files = modified_files.union(all_files_edited_in_pack_ignore)
```
Comment (JudahSchwartz) [inline demisto_sdk/commands/validate/validate_manager.py:2042]: comment needed?

**Review by JudahSchwartz — APPROVED** [review]

File: demisto_sdk/commands/validate/validate_manager.py (hunk 10 above)
Comment (JudahSchwartz) [reply demisto_sdk/commands/validate/validate_manager.py:2770]: lets not use magic numbers wherever possible

Comment (dorschw) [reply demisto_sdk/commands/validate/validate_manager.py:2770]:
```suggestion
                file_name = section[len("files"):]
```

File: demisto_sdk/commands/validate/validate_manager.py (hunk 11 above)
Comment (YuvHayun) [reply demisto_sdk/commands/validate/validate_manager.py:2018]: I tried to search tools.py but couldn't find anything..

**Pull request merged**